src/annotation.js

```diff
@@ -170,7 +182,7 @@ function draw(chart, options, caller) {
 	clipArea(ctx, chartArea);
 	for (let i = 0; i < elements.length; i++) {
 		const el = elements[i];
-		if ((el.options.drawTime || options.drawTime || caller) === caller) {
+		if ((el.options.drawTime || options.drawTime || caller) === caller && el._display) {
```

Comment:
Should probably check the `_display` first, might save a cpu cycle there :)

Reply:
pushed updated PR